invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -86,6 +90,22 @@ def _dump_resource_type_vocabulary():
     ]
 
 
+def _dump_linkable_resource_type_vocabulary():
+    """Dump linkable resource type vocabulary."""
+    return _dump_resource_type_vocabulary(
+        extra_filter=~Q('exists', field="props.invenio_type") |
+        Q('term', props__invenio_type="linkable")
+    )
+
+
+def _dump_depositable_resource_type_vocabulary():
+    """Dump depositable resource type vocabulary."""
+    return _dump_resource_type_vocabulary(
+        extra_filter=~Q('exists', field="props.invenio_type") |
+        Q('term', props__invenio_type="depositable")
```

Comment:
It's a little strange to me that at the end of the day we have to construct an ElasticSearch query in the UI views code to get the resource_types... (I understand some of the context... still perhaps valuable to revisit)

Reply:
I agree with this. I think it sort of breaks the resources/service paradigm. The filter should be somehow constructed at service level and this layer (view) use said function. Maybe a function is not the way and it can be an attribute or something idk. But I agree it is weird to see a elasticsearch query instantiation in the UI layer.

Reply:
I see your point but I cannot think of a nice a solution. 
I would say that we can crate a new folder in [`invenio-rdm-records/services`](https://github.com/inveniosoftware/invenio-rdm-records/tree/master/invenio_rdm_records/services) called vocabularies and there we can build the specific queries to retrieve both cases. WDYT?
(any suggestion is welcomed)